src/include/execution/compiler/pipeline.h

```diff
@@ -241,31 +259,51 @@ class Pipeline {
   /** @return The vector of pipeline operators that make up the pipeline. */
   const std::vector<OperatorTranslator *> &GetTranslators() const { return steps_; }
 
+  void InjectStartPipelineTracker(FunctionBuilder *builder) const;
+
+  void InjectEndResourceTracker(FunctionBuilder *builder, query_id_t query_id) const;
+
+  ast::Identifier GetRunPipelineFunctionName() const;
+
+  void CollectDependencies(std::vector<const Pipeline *> *deps) const;
+  ast::Identifier GetTeardownPipelineFunctionName() const;
+  ast::Identifier GetInitPipelineFunctionName() const;
+
+  const StateDescriptor &GetPipelineStateDescriptor() const { return nested_ ? parent_->state_ : state_; }
+
+  StateDescriptor &GetPipelineStateDescriptor() { return nested_ ? parent_->state_ : state_; }
+
  private:
   // A unique pipeline ID.
   uint32_t id_;
   // The compilation context this pipeline is part of.
   CompilationContext *compilation_context_;
   // The code generation instance.
   CodeGen *codegen_;
+  // Cache of common identifiers.
```

Comment:
Maybe move these back to their original location?

Reply:
Clang-tidy complained about too much padding with the original arrangement